### PR TITLE
Add Keycloak support.

### PR DESCRIPTION
**Pull request by gildur**
Authentication via Keycloak(http://www.keycloak.org/).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ Supported OAuth providers:
 * [GitHub](https://developer.github.com/v3/oauth/)
 * [GitLab](https://about.gitlab.com/)
 * [Google](https://developers.google.com/identity/protocols/OAuth2)
+* [Keycloak](http://www.keycloak.org/)
 
 See the [Wiki](https://github.com/davido/gerrit-oauth-provider/wiki) what it can do for you.
```
Comment (kalenobr) [inline README.md]: 1.0 version

**Comment by davido** [comment]
Thanks for the contribution. Can you split refactoring from new feature? Can you upload these changes to the gerrit-review, as mentioned in the description of this repository?

Please upload changes for review to: https://gerrit-review.googlesource.com/#/admin/projects/plugins/oauth.

**Comment by gildur** [comment]
I split refactoring from new feature and uploaded two commits to gerrit-review:
https://gerrit-review.googlesource.com/#/c/plugins/oauth/+/147291/
https://gerrit-review.googlesource.com/#/c/plugins/oauth/+/147292/

**Comment by davido** [comment]
> I split refactoring from new feature and uploaded two commits to gerrit-review:
https://gerrit-review.googlesource.com/#/c/plugins/oauth/+/147291/
https://gerrit-review.googlesource.com/#/c/plugins/oauth/+/147292/

Thanks. These CLs were merged.

**Comment by davido** [comment]
I released [new version](https://github.com/davido/gerrit-oauth-provider/releases/tag/v2.14.6) that includes these CLs.

**Comment by mwebber** [comment]
I tested the new version on our system (Gerrit 2.14.6) which uses CAS authentication, and I can confirm that it works OK. So although that doesn't test the new feature, it does confirm that the refactor did not break anything, at least for us.

**Comment by davido** [comment]
Thanks @mwebber!

**Pull request closed without merging**